apps/site/src/app/admin/participants/components/ParticipantAction.tsx

```diff
@@ -47,7 +50,27 @@ function ParticipantAction({
 		</Button>
 	);
 
-	if (participant.status === Status.waitlisted) {
+	const confirmButton = (
+		<Button
+			variant="inline-link"
+			ariaLabel={`Confirm attendance for ${participant._id}`}
+			onClick={() => initiateConfirm(participant)}
+			disabled={!isCheckin}
+		>
+			Confirm
+		</Button>
+	);
+
+	if (nonHacker) {
+		if (role !== "director") {
+			return (
+				<ParticipantActionPopover content="Only directors are allowed to confirm non-hackers.">
+					{confirmButton}
+				</ParticipantActionPopover>
+			);
+		}
+		return confirmButton;
```

Comment:
Issue: confirmation button should be shown only if status is `WAIVER_SIGNED`, should be allowed to check-in otherwise.